File: Euler 0009 - Special Pythagorean Theorem/euler_0009.py
# https://projecteuler.net/problem=9
# created 1.11.2018 by CB Fay

from math import sqrt

def solution(target):
    for a in range(1, target//2):
        for b in range(a, target//2):
            c = sqrt(a**2 + b**2)
            if a**2 + b**2 == c**2 and a + b + c == 1000:
                print(a, b, c)
                print(a * b * c)
                exit()

solution(1000)

File: Euler 0003 - Largest Prime Factor/euler_0009.scala
// https://projecteuler.net/problem=9
// created 1.11.2018 by CB Fay

import math.{sqrt, pow}

object euler_0009 {
    def main(args : Array[String]) {
        var a : Double = 1
        var b : Double = 1
        var c : Double = 1
        while (a < 500) {
            a += 1
            b = a + 1
            while (b < 500) {
                b += 1
                c = sqrt(pow(a, 2) + pow(b, 2))
                if (a + b + c == 1000) {
                    println(a, b, c)
                    println((a * b * c).toLong)
                }
            }
        }
    }
}
